### PR TITLE
Diagnostic notification logging filter

### PR DESCRIPTION
**Pull request by marcschier**

### CODE REVIEW

**Review by mregen — DISMISSED** [review]

**Review by mregen — DISMISSED** [review]
:shipit:

File: src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs [hunk 1]
```diff
@@ -269,8 +269,18 @@ private async Task UpdateApiKeyAndCertificateAsync()
...
+                var san = new SubjectAlternativeNameBuilder();
+                san.AddDnsName(dnsName);
+                await AddAddressesAsync(dnsName, san).ConfigureAwait(false);
 
+                var altDns = _identity?.ModuleId ?? _identity?.DeviceId;
+                if (!string.IsNullOrEmpty(altDns))
+                {
+                    san.AddDnsName(altDns);
+                    await AddAddressesAsync(altDns, san).ConfigureAwait(false);
+                }
```
Comment (mregen) [inline src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs:293]: I would also check the dns names on existing certs to see if they still match, unless it is safe to assume they are always the same

File: src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs [hunk 2]
```diff
@@ -289,6 +299,19 @@ private async Task UpdateApiKeyAndCertificateAsync()
                 "Stored new Certificate in {Store} store (and scheduled renewal after {Duration}).",
                 apiKeyStore.Name, renewalDuration);
             _certificateRenewals++;
+
+            async Task AddAddressesAsync(string hostName, SubjectAlternativeNameBuilder san)
+            {
+                try
+                {
+                    var addresses = await Dns.GetHostAddressesAsync(hostName).ConfigureAwait(false);
+                    addresses.ForEach(san.AddIpAddress);
```
Comment (mregen) [inline src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs]: >san.AddIpAddress [](http://example.com/codeflow?start=38&length=16)

does it filter duplicates?

File: src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs [hunk 3]
```diff
@@ -289,6 +299,19 @@ private async Task UpdateApiKeyAndCertificateAsync()
                 "Stored new Certificate in {Store} store (and scheduled renewal after {Duration}).",
                 apiKeyStore.Name, renewalDuration);
             _certificateRenewals++;
+
+            async Task AddAddressesAsync(string hostName, SubjectAlternativeNameBuilder san)
+            {
+                try
+                {
+                    var addresses = await Dns.GetHostAddressesAsync(hostName).ConfigureAwait(false);
```
Comment (mregen) [inline src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs:319]: >Dns.GetHostAddressesAsync(h [](http://example.com/codeflow?start=42&length=27)

does it also return Ipv6 temporary addresses, not sure if there is a need to filter

File: src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs (hunk 2 above)
Comment (marcschier) [reply src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs]: Good one, added filter.

File: src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs (hunk 3 above)
Comment (marcschier) [reply src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs:319]: Should it filter out all ipv6?  Or just some

Comment (mregen) [reply src/Azure.IIoT.OpcUa.Publisher/src/Services/RuntimeStateReporter.cs:319]: if any I would only add global unicast IPv6 addresses (2000::/3) or no IPv6 at all.

**Review by mregen — DISMISSED** [review]
:shipit:

**Pull request closed without merging**